enlighten/measurement/MeasurementFactory.py

```diff
@@ -88,21 +90,31 @@ def create_from_spectrometer(self, spec, is_collapsed, generate_thumbnail=True,
         log.debug("creating Measurement from spec %s", spec.label)
 
         # instantiate the Measurement
-        measurement = Measurement(
-            save_options = self.save_options,
-            spec         = spec,
-            measurements = self.measurements)
+        try:
+            measurement = Measurement(
+                save_options = self.save_options,
+                spec         = spec,
+                measurements = self.measurements)
+        except:
+            msgbox("Failed to create measurement", "Error")
```

Comment:
Note that if you add a log.error(exc_info=1) we'll get the stacktrace.

Honestly, I sometimes include stacktraces in on-screen displays...makes it easier for the user to just send me a screenshot.

Reply:
Nice idea. I've implemented this without modifying msgbox, using the traceback module